includes/admin/tools/class-settings-import.php

```diff
@@ -0,0 +1,681 @@
+						<?php
+					}
+					?>
+                </optgroup>
+
+                <optgroup label="Donors">
+					<?php
+					foreach ( $donors as $option => $option_value ) {
+						$checked = self::selected( $option_value, $value );
+						?>
+                        <option value="<?php echo $option; ?>" <?php echo $checked; ?> ><?php echo $option_value; ?></option>
+						<?php
+					}
+					?>
+                </optgroup>
+
+                <optgroup label="Forms">
+					<?php
+					foreach ( $forms as $option => $option_value ) {
+						$checked = self::selected( $option_value, $value );
+						?>
+                        <option value="<?php echo $option; ?>" <?php echo $checked; ?> ><?php echo $option_value; ?></option>
+						<?php
+					}
+					?>
+                </optgroup>
+
+				<?php
+				do_action( 'give_import_dropdown_option', $index, $donations, $donors, $forms, $value );
+				?>
+            </select>
+			<?php
+		}
+
+		static function get_csv_total( $file_id ) {
+			$total = false;
+			if ( $file_id ) {
+				$file_dir = get_attached_file( $file_id );
+				if ( $file_dir ) {
+					$file = new SplFileObject( $file_dir, 'r' );
+					$file->seek( PHP_INT_MAX );
+					$total = $file->key() + 1;
+				}
+			}
+
+			return $total;
+		}
+
+		static function get_importer( $file_id, $index = 0, $delimiter = ',' ) {
+			$raw_data = false;
+			$file_dir = get_attached_file( $file_id );
+			if ( $file_dir ) {
+				if ( false !== ( $handle = fopen( $file_dir, 'r' ) ) ) {
+					$raw_data = fgetcsv( $handle, $index, $delimiter );
+					// Remove BOM signature from the first item.
+					if ( isset( $raw_data[0] ) ) {
+						$raw_data[0] = self::remove_utf8_bom( $raw_data[0] );
+					}
+				}
+			}
+
+			return $raw_data;
+		}
+
+		/**
+		 * Remove UTF-8 BOM signature.
+		 *
+		 * @param  string $string String to handle.
+		 *
+		 * @return string
+		 */
+		static function remove_utf8_bom( $string ) {
+			if ( 'efbbbf' === substr( bin2hex( $string ), 0, 6 ) ) {
+				$string = substr( $string, 3 );
+			}
+
+			return $string;
+		}
+
+
+		/**
+		 * Is used to show the process when user upload the donor form.
+		 *
+		 * @since 1.2
+		 */
+		static function progress() {
+			$step = Give_Settings_Import::get_step();
+			?>
+            <ol class="give-progress-steps">
+                <li class="<?php echo( 1 === $step ? 'active' : '' ); ?>">Upload CSV file</li>
+                <li class="<?php echo( 2 === $step ? 'active' : '' ); ?>">Column mapping</li>
+                <li class="<?php echo( 3 === $step ? 'active' : '' ); ?>">Import</li>
+                <li class="<?php echo( 4 === $step ? 'active' : '' ); ?>">Done!</li>
+            </ol>
+			<?php
+		}
+
+		/**
+		 * Will return the import step.
+		 *
+		 * @since 1.2
+		 *
+		 * @return int $step on which step doest the import is on.
+		 */
+		static function get_step() {
+			$step    = (int) ( isset( $_REQUEST['step'] ) ? give_clean( $_REQUEST['step'] ) : false );
```

Comment:
@raftaar1191 if accept the only int then why we set the default value to `false`. This should be `0`, right?

Reply:
@ravinderk did not get you, Can you please explain it again

Reply:
Got it. Fixed!